tests/phpunit/integration/api/actions/beansAddAction.php

```diff
@@ -0,0 +1,184 @@
+
+		// Now check that it was registered in WordPress.
+		$this->assertTrue( has_action( $this->action['hook'] ) );
+		$this->check_parameters_registered_in_wp( $this->action );
+
+		// Now check in Beans.
+		$this->assertEquals( $this->action, _beans_get_action( 'foo', 'added' ) );
+	}
+
+	/**
+	 * Test beans_add_action() should use the action configuration in "replaced" status, when it's available.
+	 */
+	public function test_should_use_replaced_action_when_available() {
+		$this->check_not_added( 'foo', $this->action['hook'] );
+
+		// Setup by storing in the "replaced" status.
+		$replaced_action = array(
+			'hook'     => $this->action['hook'],
+			'callback' => 'my_callback',
+			'priority' => 20,
+			'args'     => 2,
+		);
+		_beans_set_action( 'foo', $replaced_action, 'replaced', true );
+
+		// Add the action.
+		$this->assertTrue( beans_add_action( 'foo', $this->action['hook'], $this->action['callback'] ) );
+
+		// Now check that it was registered in WordPress.
+		$this->assertTrue( has_action( $replaced_action['hook'] ) );
+		$this->check_parameters_registered_in_wp( $replaced_action );
+
+		// Now check in Beans.
+		$this->assertEquals( $replaced_action, _beans_get_action( 'foo', 'added' ) );
+	}
+
+	/**
+	 * Test beans_add_action() should return null when the ID is registered to the "removed" status.
+	 */
+	public function test_should_return_null_when_removed() {
+		$this->check_not_added( 'foo', $this->action['hook'] );
+
+		// Setup by storing in the "removed" status.
+		_beans_set_action( 'foo', $this->action, 'removed', true );
+
+		// Add the action.
+		$this->assertNull( beans_add_action( 'foo', $this->action['hook'], $this->action['callback'] ) );
+
+		// Now check that it was registered in WordPress.
+		$this->assertFalse( has_action( $this->action['hook'] ) );
+		global $wp_filter;
+		$this->assertFalse( array_key_exists( $this->action['hook'], $wp_filter ) );
+
+		// Now check in Beans.
+		$this->assertEquals( $this->action, _beans_get_action( 'foo', 'added' ) );
+	}
+
+	/**
+	 * Test beans_add_action() should merge the "modified" action configuration parameters.
+	 */
+	public function test_should_merge_modified_action_parameters() {
+		$this->check_not_added( 'foo', $this->action['hook'] );
+
+		// Setup by storing in the "modified" status.
+		$modified_action = array(
+			'hook'     => $this->action['hook'],
+			'callback' => 'my_callback',
+			'priority' => 20,
+			'args'     => 2,
+		);
+		_beans_set_action( 'foo', $modified_action, 'modified', true );
+
+		// Add the action.
+		$this->assertTrue( beans_add_action( 'foo', $this->action['hook'], $this->action['callback'] ) );
+
+		// Now check that it was registered in WordPress.
+		$this->assertTrue( has_action( $modified_action['hook'] ) );
+		$this->check_parameters_registered_in_wp( $modified_action );
+
+		// Now check in Beans.
+		$this->assertEquals( $this->action, _beans_get_action( 'foo', 'added' ) );
+	}
+
+	/**
+	 * Check that is not registered first.
```

Comment:
Maybe make comment and function name consistent... 
`Check that the action is not registered first.`
`protected function check_action_not_registered` 
This is the only time/test we 'check' instead of 'test'